polaris-react/src/components/Scrollable/Scrollable.tsx

```diff
@@ -38,103 +45,127 @@ export interface ScrollableProps extends React.HTMLProps<HTMLDivElement> {
   onScrolledToBottom?(): void;
 }
 
-export function Scrollable({
-  children,
-  className,
-  horizontal = true,
-  vertical = true,
-  shadow,
-  hint,
-  focusable,
-  onScrolledToBottom,
-  ...rest
-}: ScrollableProps) {
-  const [topShadow, setTopShadow] = useState(false);
-  const [bottomShadow, setBottomShadow] = useState(false);
-  const stickyManager = useLazyRef(() => new StickyManager());
-  const scrollArea = useRef<HTMLDivElement>(null);
-  const scrollTo = useCallback((scrollY: number) => {
-    const behavior = prefersReducedMotion() ? 'auto' : 'smooth';
-    scrollArea.current?.scrollTo({top: scrollY, behavior});
-  }, []);
-
-  const handleScroll = useCallback(() => {
-    const currentScrollArea = scrollArea.current;
-
-    if (!currentScrollArea) {
-      return;
-    }
+export interface ScrollToOptions {
+  behavior?: 'instant' | 'smooth' | 'auto';
+}
 
-    requestAnimationFrame(() => {
-      const {scrollTop, clientHeight, scrollHeight} = currentScrollArea;
-      const canScroll = Boolean(scrollHeight > clientHeight);
-      const isBelowTopOfScroll = Boolean(scrollTop > 0);
-      const isAtBottomOfScroll = Boolean(
-        scrollTop + clientHeight >= scrollHeight - LOW_RES_BUFFER,
-      );
-
-      setTopShadow(isBelowTopOfScroll);
-      setBottomShadow(!isAtBottomOfScroll);
-
-      if (canScroll && isAtBottomOfScroll && onScrolledToBottom) {
-        onScrolledToBottom();
+export interface ScrollableRef {
+  scrollTo: (scrollY: number, options: ScrollToOptions) => void;
+}
+
+const ScrollableComponent = forwardRef<ScrollableRef, ScrollableProps>(
+  (
+    {
+      children,
+      className,
+      horizontal = true,
+      vertical = true,
+      shadow,
+      hint,
+      focusable,
+      onScrolledToBottom,
+      ...rest
+    }: ScrollableProps,
+    forwardedRef,
+  ) => {
+    const [topShadow, setTopShadow] = useState(false);
+    const [bottomShadow, setBottomShadow] = useState(false);
+    const stickyManager = useLazyRef(() => new StickyManager());
+    const scrollArea = useRef<HTMLDivElement>(null);
+
+    const scrollTo = useCallback(
+      (scrollY: number, options: ScrollToOptions = {}) => {
+        const optionsBehavior = options.behavior || 'smooth';
+        const behavior = prefersReducedMotion() ? 'auto' : optionsBehavior;
+        // @ts-expect-error TS removed "instant" option but browsers support it.
+        scrollArea.current?.scrollTo({top: scrollY, behavior});
+      },
+      [],
+    );
+
+    const defaultRef = useRef();
```

Comment:
Curious, what happens when a `defaultRef` isn't provided?

Reply:
Test failures, mostly. useImperativeHandle throws errors trying to set `current` of null.

Reply:
Interesting, thanks for catching this!

Do you have an example, just for my own understanding? It's not clear from the [React documentation](https://beta.reactjs.org/reference/react/useImperativeHandle) that this would be necessary and just curious to know how this fixes the error.